cmd/influx/write.go

```diff
@@ -11,271 +11,237 @@ import (
-			},
-		}
+	ErrorsFile    string
+	MaxLineLength int
+	RateLimit     throttler.BytesPerSec
 
-		return cli.Write(standardCtx(ctx), writeClients, &internal.WriteParams{
-			BucketID:   ctx.String("bucket-id"),
-			BucketName: ctx.String("bucket"),
-			OrgID:      ctx.String("org-id"),
-			OrgName:    ctx.String("org"),
-			Precision:  precision,
-		})
-	},
-	Subcommands: []*cli.Command{
-		{
-			Name:        "dryrun",
-			Usage:       "Write to stdout instead of InfluxDB",
-			Description: "Write protocol lines to stdout instead of InfluxDB. Troubleshoot conversion from CSV to line protocol",
-			Flags:       writeFlags,
-			Action: func(ctx *cli.Context) error {
-				format, err := parseFormat(ctx.String("format"))
-				if err != nil {
-					return err
-				}
-				compression, err := parseCompression(ctx.String("compression"))
-				if err != nil {
-					return err
-				}
+	internal.WriteParams
+}
 
-				var errorOut io.Writer
-				if ctx.IsSet("errors-file") {
-					errorFile, err := os.Open(ctx.String("errors-file"))
-					if err != nil {
-						return fmt.Errorf("failed to open errors-file: %w", err)
-					}
-					defer errorFile.Close()
-					errorOut = errorFile
-				}
+func (p *writeParams) makeLineReader(args []string, errorOut io.Writer) *linereader.MultiInputLineReader {
+	return &linereader.MultiInputLineReader{
+		StdIn:                      os.Stdin,
+		HttpClient:                 http.DefaultClient,
+		ErrorOut:                   errorOut,
+		Args:                       args,
+		Files:                      p.Files.Value(),
+		URLs:                       p.URLs.Value(),
+		Format:                     p.Format,
+		Compression:                p.Compression,
+		Encoding:                   p.Encoding,
+		Headers:                    p.Headers.Value(),
+		SkipRowOnError:             p.SkipRowOnError,
+		SkipHeader:                 p.SkipHeader,
+		IgnoreDataTypeInColumnName: p.IgnoreDataTypeInColumnName,
+		Debug:                      p.Debug,
+	}
+}
 
-				cli, err := newCli(ctx)
-				if err != nil {
-					return err
-				}
-				reader := &linereader.MultiInputLineReader{
-					StdIn:                      os.Stdin,
-					HttpClient:                 http.DefaultClient,
-					ErrorOut:                   errorOut,
-					Args:                       ctx.Args().Slice(),
-					Files:                      ctx.StringSlice("file"),
-					URLs:                       ctx.StringSlice("url"),
-					Format:                     format,
-					Compression:                compression,
-					Encoding:                   ctx.String("encoding"),
-					Headers:                    ctx.StringSlice("header"),
-					SkipRowOnError:             ctx.Bool("skipRowOnError"),
-					SkipHeader:                 ctx.Int("skipHeader"),
-					IgnoreDataTypeInColumnName: ctx.Bool("xIgnoreDataTypeInColumnName"),
-					Debug:                      ctx.Bool("debug"),
-				}
+func (p *writeParams) makeErrorFile(ctx *cli.Context) (*os.File, error) {
+	if !ctx.IsSet("errors-file") {
```

Comment:
The errors-file flag is wired up to a struct field, so could we change this to:
```suggestion
func (p *writeParams) makeErrorFile() (*os.File, error) {
	if p.ErrorsFile == "" {
```